### PR TITLE
RavenDB-19839 Create Windows docker images for LTSC 2022

### PR DESCRIPTION
**Pull request by poissoncorp**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-19839/Create-Windows-docker-images-for-LTSC-2022

### Additional description

I've created a new Dockerfile that is runnable on ltsc2022 kernel windows machines.
Modified the build, the publish and other common scripts.

### Type of change

- New feature

### How risky is the change?

- Low 

### Backward compatibility
- Not relevant

### Is it platform specific issue?

- Yes. Windows Nanoserver

### Documentation update

- This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.

### Testing by Contributor

- It has been verified by manual testing

### Testing by RavenDB QA team

- No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- Yes. There's a new argument called $WinVer in common & publish files. It can be set to 1809 or 2022

### CODE REVIEW

File: docker/ravendb-nanoserver/Dockerfile.2022
Comment (poissoncorp) [inline docker/ravendb-nanoserver/Dockerfile.2022]: lmk if F is ok @gregolsky

File: docker/build-nanoserver.ps1
Comment (gregolsky) [inline docker/build-nanoserver.ps1]: cant default to new one. need to default to the previous one

File: docker/common.ps1
Comment (gregolsky) [inline docker/common.ps1]: add 1809 tag please

Comment (gregolsky) [reply docker/common.ps1]: do we want 2022 as `latest`? that would make sense, but it might break people's servers. Maybe we should drop `latest` and `windows-latest` altogether?

cc @ppekrol

File: docker/ravendb-nanoserver/Dockerfile.2022
Comment (gregolsky) [reply docker/ravendb-nanoserver/Dockerfile.2022]: I think so. Does M work as in 1809?

File: docker/common.ps1
Comment (poissoncorp) [reply docker/common.ps1]: I think that the best way is to leave it like this.

We should have the `latest` tag, it feels like a standard, but the kernels across different windows versions are various. 

Many people could have their pipelines set up using this floating docker tag. If they're using a system that isn't working on 2022 kernel, it's gonna break it.

Comment (poissoncorp) [reply docker/common.ps1]: but after all, now it looks like this is the one and only opportunity to delete the 'latest' tag

Comment (gregolsky) [reply docker/common.ps1]: I think we should deprecate it - stop pushing new images there. `*latest*` should be gone including `5.4-windows-latest`. Instead we should have
`5.4-windows-lts1809` and `5.4-windows-ltsc2022` and so on. Let's wait for approval from @ppekrol though.
There are also CI/CD challenges ahead here.

Comment (gregolsky) [reply docker/common.ps1]: @poissoncorp Talked with Paweł. List of tags we need:

v5.4 images

for 1809	
	
"$($repo):$($version)-windows-1809",
"$($repo):windows-1809-latest",
"$($repo):windows-1809-latest-lts",
"$($repo):5.4-windows-1809-latest"

for ltsc2022

"$($repo):$($version)-windows-ltsc2022",
"$($repo):windows-ltsc2022-latest",
"$($repo):windows-ltsc2022-latest-lts",
"$($repo):5.4-windows-ltsc2022-latest"

We're not going to update latest and latest-lts anymore

**Review by gregolsky — CHANGES_REQUESTED** [review]
See my comment about tags please

File: docker/common.ps1
Comment (poissoncorp) [reply docker/common.ps1]: removed the latest tags
added tags mentioned above

File: docker/ravendb-nanoserver/Dockerfile.2022
Comment (poissoncorp) [reply docker/ravendb-nanoserver/Dockerfile.2022]: Yes, it should.

Comment (poissoncorp) [reply docker/ravendb-nanoserver/Dockerfile.2022]: As far as I know, it does.

File: docker/ravendb-nanoserver/Dockerfile.2022
Comment (gregolsky) [inline docker/ravendb-nanoserver/Dockerfile.2022]: can we change the filename suffix to actual win version suffix `ltsc2022`

File: docker/publish-nanoserver.ps1
Comment (gregolsky) [inline docker/publish-nanoserver.ps1]: let's use `ltsc2022` and the default should be the old one

File: docker/common.ps1
Comment (gregolsky) [inline docker/common.ps1]:
```suggestion
        "ltsc2022" {
```

File: docker/publish-nanoserver.ps1
Comment (poissoncorp) [reply docker/publish-nanoserver.ps1]: done

File: docker/common.ps1
Comment (poissoncorp) [reply docker/common.ps1]: changed

File: docker/ravendb-nanoserver/Dockerfile.2022
Comment (poissoncorp) [reply docker/ravendb-nanoserver/Dockerfile.2022]: renamed the Dockerfile to Dockerfile.ltsc2022

**Review by gregolsky — APPROVED** [review]

**Pull request merged**
